engine/engine.go

```diff
@@ -152,28 +154,48 @@ func DownloadEngine(ctx context.Context, opts *options.TerragruntOptions) error
 		return nil
 	}
 	downloadFile := filepath.Join(path, enginePackageName(e))
-	var downloadURL string
+
+	downloads := make(map[string]string)
+	checksumFile := ""
+	checksumSigFile := ""
 	if strings.HasPrefix(e.Source, "http") {
```

Comment:
```suggestion
	if strings.HasPrefix(e.Source, "://") {
```

Since `go-getter` supports not only `http`, I think this expands possible sources.  
_Not sure of the correct suggestion, requires rechecking_